### PR TITLE
its 4:10am help me

### PR DESCRIPTION
**Pull request by Rick10101221**
math midterm tmmr havent studied :)

Changes:
- All routes are filled-in (with the exception of corkboard and hugs, whose models need to be completed and tested)
- Added TODO tags on the top of routes (in users, notifications, and hugs) that are broken and VERIFIED tags for ones that are fully completed and operational
- Added a fully-fledged suggestion of code in buildCorkboard that will synchronously build the corkboard (with the exception of the date_time field, which cannot be converted from TimeStamp Firestore object to JSON, so it is not being passed back at all. This needs to be looked into)

### CODE REVIEW

File: backend/model/Notifications.js [hunk 1]
```diff
@@ -50,40 +30,40 @@ const NotificationsAPI = {
...
+        // Gets the time that the notification is sent
+        var dateTime = new Date();
+        
+        const newHugCollectionRef = 
+          users.doc(friend_id).collection('notifications').doc(`${hug_id}`);
+
+        const newHug = {
+          type : "hug",
+          hug_id : hug_id,
+          date_time : dateTime,
```
Comment (terryzfeng) [inline backend/model/Notifications.js:75]: This needs to be changed to a firebase timestamp object

File: backend/routes/notificationsRoutes.js [hunk 2]
```diff
@@ -53,51 +54,85 @@ async function notificationCollectionLength(req, uid) {
...
+    // let users = []; // { friendName, friendPFP }
+    for (let i = 0; i < notifResponse.length; i++) {
+      const userId = notifResponse[i].friend;
+      const userResponse = await UsersAPI.getUserProfile(userId);
+      const newUser = {
+        friendName: userResponse.name,
+        friendPfp: userResponse.profile_pic
+      }
+      notifResponse[i]['friendInfo'] = newUser;
+    }
```
Comment (terryzfeng) [inline backend/routes/notificationsRoutes.js]: We should be doing this work in the body of the function, not the route

File: backend/model/Corkboard.js [hunk 3]
```diff
@@ -8,47 +8,82 @@ require("firebase/auth");
...
+const hugs = db.collection('hugs');
+
 const CorkboardAPI = {
 
-    buildCorkboard: function() {
-        const currUser = firebase.auth().currentUser;
+    buildCorkboard: async function(uid) {
+        // const currUser = firebase.auth().currentUser;
+
         let pinnedHugsList = [];
```
Comment (terryzfeng) [inline backend/model/Corkboard.js:18]: @EmanSherif

File: backend/model/Notifications.js [hunk 4]
```diff
@@ -11,37 +11,17 @@ const users = db.collection("users");
...
-                .limit(5);
+        // UNPAGINATED
+        var notificationCollection = users
+          .doc(uid)
+          .collection("notifications")
+          .orderBy("date_time")
+        const notificationSnapshot = await notificationCollection.get();
+        let notifications = [];
+        notificationSnapshot.forEach(doc => {
+          notifications = [...notifications, doc.data()];
```
Comment (terryzfeng) [inline backend/model/Notifications.js:26]: Notifications should be handling the logic, so the route is just calling this and passing back the return

File: backend/model/Users.js [hunk 5]
```diff
@@ -84,6 +84,16 @@ const UsersAPI = {
...
+  checkUserExists: async function(email) {
+    let success = false;
+    const userRef = usersCollection.where('email', '==', email);
+    const user = await userRef.get();
+    if (!user.exists) {
+      success = true;
+    }
+    return { out: status };
+  },
+
```
Comment (evanjserrano) [inline backend/model/Users.js]: Good function. Need to update documentation

**Review by evanjserrano — APPROVED** [review]

File: backend/model/Users.js (hunk 5 above)
Comment (terryzfeng) [reply backend/model/Users.js]: Do we have an email field?

Comment (evanjserrano) [reply backend/model/Users.js]: Oh oops we're not storing the email so this should be with something else.. maybe use authentication? I can work on this if you need

Comment (evanjserrano) [reply backend/model/Users.js]: I take it back. Bad function. ;) jk

**Review by evanjserrano — DISMISSED** [review]
Fix or change that Users.js func

**Comment by Rick10101221** [comment]
Changes
- Scratched checkUserExists(email)
- Notifications now have firestore timestamp objects instead of JS Date objects
- Moved notificationRoutes.js model logic from routes to Notifications.js
- Added route skeletons to friendsRoutes.js in prep for Friends.js model merge
- Added NOT TESTED and BROKEN tags at the top of routes for Friends, Corkboards, Accounts

**Review by evanjserrano — APPROVED** [review]

**Review by terryzfeng — APPROVED** [review]
Just realized this was from the backend/hugs branch. All good from me.

**Review by tyusliu — APPROVED** [review]
I see what you did in Hugs, lgtm. Only thing is in hugsRoutes the comment says we are still trying to figure out blobs? or we sticking with base 64 something?

File: backend/model/Users.js [hunk 6]
```diff
@@ -139,6 +139,7 @@ const UsersAPI = {
   },
 
   uploadUserProfilePicture: async function (uid, file) {
+    console.log('here');
```
Comment (vukRadovanovic) [inline backend/model/Users.js:142]: I believe that whatever message is logged should be a little bit more descriptive.

**Review by vukRadovanovic — COMMENTED** [review]
Looks good!  Just that minor detail of what's being console.logged in Users.js that can be easily resolved.

**Pull request merged**